libs-sol/nearbridge/contracts/NearBridge.sol

```diff
@@ -123,32 +125,23 @@ contract NearBridge is Ownable {
             );
         }
 
-        // 4. approvals_next and approvals_after_next contain signatures that check out against the block producers for the epoch of the block
-        // 5. The signatures present in both approvals_next and approvals_after_next correspond to more than 2/3 of the total stake
-        // uint256 totalStake = 0;
-        // for (uint i = 0; i < nearBlock.next_bps.validatorStakes.length; i++) {
-        //     totalStake = totalStake.add(
-        //         nearBlock.next_bps.validatorStakes[i].stake
-        //     );
-        // }
-        // require(
-        //     _checkValidatorSignatures(
-        //         totalStake,
-        //         nearBlockHash,
-        //         nearBlock.approvals_next,
-        //         nearBlock.next_bps.validatorStakes
-        //     ),
-        //     "NearBridge: Less than 2/3 voted by the next block"
-        // );
-        // require(
-        //     _checkValidatorSignatures(
-        //         totalStake,
-        //         nearBlockNextHash,
-        //         nearBlock.approvals_after_next,
-        //         nearBlock.next_bps.validatorStakes
-        //     ),
-        //     "NearBridge: Less than 2/3 voted by the block after next"
-        // );
+        // 4. approvals_after_next contain signatures that check out against the block producers for the epoch of the block
+        // 5. The signatures present in approvals_after_next correspond to more than 2/3 of the total stake
+        uint256 totalStake = 0;
+        uint256 votedFor = 0;
+        if (prev.bps_count > 0) {
+            require(nearBlock.next_bps.validatorStakes.length == prev.bps_count, "NearBridge: number of BPs should match number of approvals");
```

Comment:
As per spec, number of approvals can be smaller than the number of BPs

Reply:
Some approvals are empty but lengths should match, isn't it?

Reply:
No, the length can be shorter, in which case it is assumed that the rest is `None`s.

It is not very noticeable in the spec, but it's there:

> approvals_after_next can contain fewer elements than next_bps in the previous epoch.

(https://github.com/nearprotocol/NEPs/blob/master/specs/ChainSpec/LightClient.md)